src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -695,6 +697,26 @@ public function appendToFile(string $filename, $content)
         }
     }
 
+    /**
+     * Watches a file or directory for any changes, and calls $callback when any changes are detected.
+     *
+     * @param mixed    $path     The path to watch for changes. Can be a path to a file or directory, iterator or array with paths
+     * @param callable $callback The callback to execute when a change is detected
+     * @param float    $timeout  The idle timeout in milliseconds after which the process will be aborted if there are no changes detected
+     *
+     * @throws \InvalidArgumentException|IOException
+     */
+    public function watch($path, callable $callback, float $timeout = null)
```

Comment:
It would be nice if we can provide our own watcher with the `WatcherInterface`.

Reply:
I think if you create your own class that implements `WatcherInterface`, then you can just use that class directly, there would be no benefit then to pass it to this method

Reply:
Yes, You're right.